docs: document the x-deprecated option (#38969)
angular/angular: aio/content/guide/schematics-authoring.md

You can see examples of schema files for the Angular CLI command schematics in [`@schematics/angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/schematics/angular/application/schema.json).

#### Deprecated options

To flag an option as deprecated, you can use the `x-deprecated` property.

> The assigned value will be used to warn the user when the deprecated option is used.

Reviewer: ```suggestion
Angular uses the assigned value when notifying the user that an option is deprecated.
```

Or something like this that removes the future tense and the passive. Is it still correct with "Angular" as the subject? I could be wrong on that one. If the subject could be more specific than Angular, that'd be even better.

Author: Thanks @kapunahelewong 
I'm currently taking the technical writer course and this is a great example of it 💯
